tests/utils/test_requirements_utils.py

```diff
@@ -697,3 +697,24 @@ def predict(self, context, model_input, params=None):
         mock_warning.assert_not_called()
         assert "pandas" in modules
         assert "sklearn" in modules
+
+
+def test_capture_imported_modules_extra_env_vars():
+    class TestModel(mlflow.pyfunc.PythonModel):
+        def predict(self, context, model_input, params=None):
+            assert os.environ["TEST"] == "test"
+            import sklearn  # noqa: F401
+
+            return model_input
+
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            "model", python_model=TestModel(), input_example="test"
+        )
+
+    with mock.patch("mlflow.utils.requirements_utils._logger.warning") as mock_warning:
```

Comment:
why do we need to mock this if it's not called?

Reply:
Because we swallow the error and log a warning instead if there're any exceptions in the subprocess https://github.com/mlflow/mlflow/blob/10c3833d2347eed8ae14c2defe2b62d85a18008b/mlflow/utils/requirements_utils.py#L418
We can mock raise_on_error instead though